core/src/main/scala/org/http4s/rho/bits/ResultMatcher.scala

```diff
@@ -7,10 +7,16 @@ import scala.reflect.runtime.universe.{ Type, WeakTypeTag }
 import scalaz.concurrent.Task
 
 
-trait ResultMatcher[-R] {
```

Comment:
I'm thinking that `ResultMatcher` might be better as a case class with `conv` as a lambda.